### PR TITLE
Support shuffling block types

### PR DESCRIPTION
**Pull request by dyceron**
Changes all the blocks of a certain type to another type. Note that will be a chaos option in Randovania, so it will not be logical, but the groundwork for it is there if that ever happens.

Relevant to #409

### CODE REVIEW

File: src/open_samus_returns_rando/misc_patches/block_patches.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        bmsbk = editor.get_file(f"maps/levels/c10_samus/{scenario_name}/{scenario_name}.bmsbk", Bmsbk)
+        blocks = bmsbk.raw["block_groups"]
+        for block in blocks:
+            block_type = block["types"][0]
+            temp_type: str = ""
+            for original_type, new_type in configuration["block_patches"].items():
+                # Check for the block type and change it to the new type
+                if original_type == block_type["block_type"] and temp_type == "":
+                    # Set the temp_type to the original_type to ensure the block doesnt get changed again
+                    temp_type = original_type
```
Comment (ThanatosGit) [inline src/open_samus_returns_rando/misc_patches/block_patches.py:20]: Please remove the `temp_type` shenanigans and just use `break` to stop the inner `for` loop.

**Review by ThanatosGit — CHANGES_REQUESTED** [review]
More general question: How is that supposed to work with the static changes?
Should this be pre- or post the static_fixes?
Does this even make a difference? The A1 teleporter crumbles doesn't really care what block type they are.

Can you destroy a crumble block in a one tile gap within a wall or will all those pickups be unacessible then?

**Comment by Miepee** [comment]
> Can you destroy a crumble block in a one tile gap within a wall or will all those pickups be unacessible then?

A ~~weight~~ crumble block gets destroyed if you touch it, doesn't necessarily have to be from above. You can just jump into them, spider boost into them, probably morph into them doesn't really matter.

As example, dcr had a test seed where the power bomb blocks here were replaced with crumbles, and they just got destroyed by jumping into them from side
grafik

**Comment by ThanatosGit** [comment]
They definitely don't get destroyed if touching them from below 😝.
Iirc there is even a small spot you could spider ball around the corner without them getting destroyed, that's why I asked explicitly about one tile gaps. But I assume it will be just fine and that you can destroy them from the side and have enough time to get the pickup before respawning.

**Comment by dyceron** [comment]
> More general question: How is that supposed to work with the static changes? Should this be pre- or post the static_fixes? Does this even make a difference? The A1 teleporter crumbles doesn't really care what block type they are.
> 
> Can you destroy a crumble block in a one tile gap within a wall or will all those pickups be unacessible then?

I guess it should be done before or not doing the static fixes at all, but then you have the ones in game patches too.

**Comment by dyceron** [comment]
I also forgot about Screw Attack blocks whoops

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**